pyphare/pyphare/pharein/restarts.py

```diff
@@ -9,20 +9,41 @@
 
 def validate(sim):
 
-    sim.restart_options["timestamps"] = phare_utilities.np_array_ify(sim.restart_options["timestamps"])
-    init_time = sim.start_time()
-
-    timestamps = sim.restart_options["timestamps"]
-    if np.any(timestamps < init_time):
-        raise RuntimeError(f"Error: timestamp({sim.time_step_nbr}) cannot be less than simulation.init_time({init_time}))")
-    if np.any(timestamps > sim.final_time):
-        raise RuntimeError(f"Error: timestamp({sim.time_step_nbr}) cannot be greater than simulation.final_time({sim.final_time}))")
-    if not np.all(np.diff(timestamps) >= 0):
-        raise RuntimeError(f"Error: {clazz}.{key} not in ascending order)")
-    if not np.all(np.abs(timestamps / sim.time_step - np.rint(timestamps/sim.time_step) < 1e-9)):
-        raise RuntimeError(f"Error: {clazz}.{key} is inconsistent with simulation.time_step")
-
-    sim.restart_options["timestamps"] = conserve_existing(sim)
+    restart_options = sim.restart_options
+
+    if "elapsed_timestamps" in restart_options:
+        import datetime
+        restart_options["elapsed_timestamps"] = [
+            int(ts.total_seconds()) if isinstance(ts, datetime.timedelta) else ts
+            for ts in phare_utilities.np_array_ify(restart_options["elapsed_timestamps"])]
+
+        if not np.all(np.diff(restart_options["elapsed_timestamps"]) >= 0):
+            raise RuntimeError(f"Error: restart_options elapsed_timestamps not in ascending order)")
+
+        seconds_in_an_hour = 60 ** 2
+        for cmp_idx, ref_ts in enumerate(restart_options["elapsed_timestamps"][1:]):
+            cmp_ts = restart_options["elapsed_timestamps"][cmp_idx]
+            if ref_ts - cmp_ts < seconds_in_an_hour:
+                raise RuntimeError(f"Error: time betweeen restart_options elapsed_timestamps must be at least one hour)")
```

Comment:
why ? Not saying we should not put limits between two checkpoints, but why 1 hour?
I guess this has to do with the fact that we want a checkpoint to be dumped **before** another one is triggered? Could that be something that breaks the code if elaped_timestamps[i+1] was already passed while the last dump is not yet even finished?

Reply:
I think I put one hour assuming it was a mistake on the users part, but it could be configurable too
Pretty sure restart dumps are synchronous so it finishes before the next simulation step starts

Reply:
left as commented out until we figure it out